### DISCUSSION TITLE
Collection items sort stopped working with rc.62

### DISCUSSION THREAD

**Discussion by domdus** [directus/directus#5577] [Archive]
Hi,

sorting of items in the app does lead to an exception since rc.62 (rc.64 also).
rc.61 is the last working one.

I tried with _npx directus database migrate:latest_ but it did not fix it.

`extensions: {code: "INTERNAL_SERVER_ERROR"}
code: "INTERNAL_SERVER_ERROR"
message: "select \"sort\", count(\"sort\") as \"count\" from \"categories\" group by \"sort\" having \"count\" > $1 - column \"count\" does not exist"` 
---

### Project details
Directus Version: 9.0.0-rc.64
Environment: production
OS: Linux 4.19.0-13-amd64
Node: 12.20.1

**Comment by rijkvanzanten** [comment]
@domdus What is your sort_field configuration for the collection? Does that field actually exist? Did you maybe delete the previous sort field while keeping the configuration intact?

**Reply by domdus** [reply]
Hi @rijkvanzanten,

yeah I checked that first - sort field exists. I created a fresh collection with sort and can reproduce the problem.
![image](https://user-images.githubusercontent.com/67951828/117671395-0b2cb200-b1a9-11eb-9614-c0546e53cf1f.png)

When enabling it does actually assign the Integer to sort field, but from then on the POST request to /utils/sort/{collection} goes HTTP 500.
![image](https://user-images.githubusercontent.com/67951828/117671567-38796000-b1a9-11eb-8fda-27f143025640.png)

**Reply by rijkvanzanten** [reply]
@domdus What database is this on?

**Reply by domdus** [reply]
It's postgresql... Still checking if it is a problem on my side when fiddling with the directus database migrate tool.

**Reply by domdus** [reply]
Checked my Host config as well, but pretty sure it's about rc.62. I can simply switch between working and non-working with

npm install directus@v9.0.0-rc.61
vs.
npm install directus@v9.0.0-rc.62

**Reply by rijkvanzanten** [reply]
What version of postgres?

**Reply by rijkvanzanten** [reply]
We did add a change that relies on the `having` SQL statement (as you can see in the error message), which should be allowed on aggregate functions (like `count`), so I'm wondering if there's a specific postgres version that doesn't play nicely with that

**Reply by domdus** [reply]
psql (PostgreSQL) 11.9 (Debian 11.9-0+deb10u1)

**Reply by rijkvanzanten** [reply]
Thanks! Turns out PG does in fact not play nicely with having on an alias select. Fixed that in aaee7cc750232ae95831a1d245eb393a0c1109a0 👍🏻

**Reply by domdus** [reply]
Hi @rijkvanzanten,

question for conclusion: Do you know if the issue with `having` strictly affects/affected postgres v11? 
Considering an upgrade to postgres 12/13.

**Reply by rijkvanzanten** [reply]
@domdus No it was a mistake on my end 👍🏻 Apparently, this doesn't work:

```sql
SELECT count(abc) AS count
FROM def
HAVING count > 1;
```

But this does:

```sql
SELECT count(abc) AS count
FROM def
HAVING count(abc) > 1;
```

👍🏻
